// base struct
message BallInfo {
    optional int32 sock = 1;
    optional int32 id = 2;
    optional double r = 3;
    optional double x = 4;
    optional double y = 5;
}
// ball's data arr
message BallsInfo {
    repeated BallInfo ball = 1;
}
// create ball cmd
message CreateBall {
    optional int32 id = 1;
    optional double x = 2;
    optional double y = 3;
}
// remove ball cmd
message RemoveBall {
    optional int32 sock = 1;
    optional int32 id = 2;
}
// remove balls
message RemoveBalls {
    repeated RemoveBall rm_ball = 1;
}
// move ball cmd
message MoveBall {
    optional int32 sock = 1;
    optional int32 id = 2;
    optional double v_val = 3;
    optional double v_dir = 4;
}
